### ISSUE TITLE
All classes are considered as the same class

### ISSUE THREAD

**Issue by NacerK** [WongKinYiu/ScaledYOLOv4#312]
Hello, 

I've been trying to finetune the pretrained yolov4-p6 using the ScaledYoloV4 large branch on a custom dataset composed of 6 classes (person, bicycle, car, moyotcycle, bus, truck). I run the following command:

 `python3 train.py --cfg ScaledYOLOv4/models/yolov4-p6.yaml --weights /weights/yolov4-p6.pt --data custom_data.yaml --batch-size 2 --epochs 50
`
The problem is, after the validations, I print the mAP of each class and it is the same value for all the classes (0.43 for each class). At the end of the training, I tried running an inference with the best.pt and all objects of interest in the image (person, bicycle, car, moyotcycle, bus, truck) are detected as the same class (person). The data is correctly labeled in the yolo .txt annotation format. 

The content of my data.yaml is: 

```
names:
- person
- bicycle
- car
- motorcycle
- bus
- truck
nc: 6
test: ''
train: jobs_dir/train_yolov4-p5_20210803151206/train.txt
val: jobs_dir/train_yolov4-p5_20210803151206/val.txt
```

Any suggestion on what is causing the problem?

Thanks in advance!

Update: after printing the value of mlc, I get 0.0 
(182 line of the train.py: `mlc = np.concatenate(dataset.labels, 0)[:, 0].max()  # max label class)`

From the comment, I understand that the variable mlc has the max label class, which in the case of 6 classes should be 5 since we start labeling from 0,1...,5. This could explain why all objects are predicted as the same class (class 0), but I don't understand where this problem is coming from. In my .txt annotations, the classes are correcly labeled from 0 to 5, I don't get why the max label class is 0 instead of 5

Update2: After exporting the visualisations during the validation (gt images and predicted images), I noticed that even on the grountruths, all the bounding boxes are classified as the class 0 (person), even though the annotations from my dataset have the correct class labels. This problem however does not happen when I an evaluation with test.py on the same dataset and export the visualisations. The gt are correctly labeled. The problem seems to be related to the way the dataset is constructed in the train.py script. 

Update3: Nevermind I found the problem. For some reason create_dataloader() is defined as follows in the datasets.py of my local code: `def create_dataloader(path, imgsz, batch_size, stride, single_cls, hyp=None, augment=False, cache=False, pad=0.0, rect=False, local_rank=-1, world_size=1):`

But in train.py, it is called this way: `dataloader, dataset = create_dataloader(train_path, imgsz, batch_size, gs, opt, hyp=hyp, augment=True,   cache=opt.cache_images, rect=opt.rect, local_rank=rank,    world_size=opt.world_size)`

it was receiving opt as a fifth parameter but was expecting as per the definition single_cls. Since opt is always True, single_cls was thus always true, and that's why all the classes were considered as a single class...
